Libraries/LibWeb/SVG/SVGDecodedImageData.cpp

```diff
@@ -120,6 +119,48 @@ RefPtr<Gfx::Bitmap> SVGDecodedImageData::render(Gfx::IntSize size) const
     return bitmap;
 }
 
+RefPtr<Gfx::PaintingSurface> SVGDecodedImageData::render_to_surface(Gfx::IntSize size) const
+{
+    VERIFY(m_document->navigable());
+
+    RefPtr<Gfx::PaintingSurface> surface;
+
+#ifdef AK_OS_MACOS
+    if (m_page_client->display_list_player_type() == DisplayListPlayerType::SkiaGPUIfAvailable) {
+        VERIFY(m_document->navigable()->skia_backend_context());
+        auto iosurface = Core::IOSurfaceHandle::create(size.width(), size.height());
+        surface = Gfx::PaintingSurface::create_from_iosurface(move(iosurface), *m_document->navigable()->skia_backend_context(), Gfx::PaintingSurface::Origin::TopLeft);
+    }
+#endif
+
+    // FIXME: Support Vulkan
+
+    if (!surface) {
+        auto bitmap = Gfx::Bitmap::create(Gfx::BitmapFormat::BGRA8888, Gfx::AlphaType::Premultiplied, size).release_value();
+        surface = Gfx::PaintingSurface::wrap_bitmap(*bitmap);
+    }
```

Comment:
since IOSurface is not shared either between processes or GPU contexts, we could simply use `PaintingSurface::create_with_size()` which will allow GPU acceleration for both Vulkan and Metal.